contracts/custom/DCNTToken.sol

```diff
@@ -0,0 +1,137 @@
+//SPDX-License-Identifier: Unlicense
+pragma solidity =0.8.19;
+
+import { IERC20Upgradeable } from "@openzeppelin/contracts-upgradeable/token/ERC20/IERC20Upgradeable.sol";
+import { MerkleProof } from "@openzeppelin/contracts/utils/cryptography/MerkleProof.sol";
+import { UndelegatableVotesERC20 } from "./UndelegatableVotesERC20.sol";
+
+contract DCNTToken is UndelegatableVotesERC20 {
```

Comment:
But regardless of which repo this contract exists in (https://github.com/decent-dao/fractal-contracts/pull/62/files#r1209304877), I'd like to talk more about this `UndelegatableVotesERC20` inheritance.

I don't want to lock this token into being "undelegatable" forever, just for now.

Earlier when I mentioned that the `delegates` override of just returning `_account` was nice and clean, I didn't realize that this code would be baked into the token contract itself (oversight on my end). 

It's very likely I have a misunderstanding of how everything is hooked together, but I have an assumption that the choice to "acknowledge" delegation lives within a "strategy" that's installed on a Fractal node. So, an ERC20 token could include code that supports delegation (standard `ERC20Votes` functionality), but when the strategy is calculating voting power, it ignores delegation and returns weight based simply on an account's balance at a block. As I type this out, I realize I have an assumption that the standard `ERC20Votes` contract has methods to not only "return an account's delegated-to voting power at some block" _and also_ "return an account's actual balance at some block".

If what I'm saying is unclear please tell me, there's lots of nuance.

Reply:
`ERC20Votes` keeps checkpoints of delegated voting power, but not individual balances.

we could definitely use or ignore delegation, and set it up so that it could be a parameter on the strategy, but we would need two checkpoints on the governance token, one for delegation, which `ERC20Votes` already does, and another historical checkpoint added for an individual's token balance

Reply:
@adamgall updated to show how that would work